### PR TITLE
support the IBMi platform

### PR DESCRIPTION
**Pull request by zheddie**
I am adding the oldest Numpy version for IBM i , which include our fixes for IBM i platform. Please help to review. thanks.

@rgommers need your help again. thx.

### CODE REVIEW

**Comment by zheddie** [comment]
if some tags needed to be bundled into the next release/tag, please also help to add it. thanks.

File: setup.cfg [hunk 1]
```diff
@@ -26,6 +26,9 @@ install_requires =
     numpy==1.16.0; python_version=='3.6' and platform_system=='AIX'
     numpy==1.16.0; python_version=='3.7' and platform_system=='AIX'
 
+    #IBM i requires fixes contained in numpy 1.23.3.
+    numpy==1.23.3; python_version=='3.9' and platform_system=='OS400'
```
Comment (rgommers) [inline setup.cfg:30]: Why only for Python 3.9? The build issue seemed independent of Python version to me.

**Review by rgommers — CHANGES_REQUESTED** [review]
Thanks @zheddie, happy to special-case OS400 here. I'll xref the `numpy.distutils` change: https://github.com/numpy/numpy/pull/22124. Also, let me note that SciPy doesn't use `oldest-supported-numpy`, so you may have to sync the relevant change to SciPy's `pyproject.toml` in case you need the fix for SciPy.

I expected CI to fail here, and I think it should because of duplicate matching requirements. The reason it doesn't is because we don't test OS400; can you add it to https://github.com/scipy/oldest-supported-numpy/blob/7c61f628ef4e10a6954d65f1ea317a85d31eac0c/tests/test_dependencies.py#L36?

File: setup.cfg (hunk 1 above)
Comment (zheddie) [reply setup.cfg:30]: Yes, we really have other python on i . Python 3.6 , but it's replaced by python 3.9 and all packages are based on python 3.9 only  now. So, it is ok we just support python 3.9 here.

**Comment by zheddie** [comment]
Curious is that SciPy can be successfully installed on IBM i now. I am assuming it is using the latest Numpy (> 1.23.3) in it's sandbox during install.  @rgommers

**Comment by zheddie** [comment]
@rgommers , any more comments ? Thanks.

**Comment by zheddie** [comment]
@rgommers can you help to review it again? thanks.

**Comment by rgommers** [comment]
The `test_has_at_most_one_pinned_dependency[CPython-3.9-OS400-s390x]` failure is kinda what I expected. Let me have a quick go at fixing that up.

File: setup.cfg (hunk 1 above)
Comment (rgommers) [reply setup.cfg:30]: There's no way for me to check this, but - in numpy you used `os400` (lowercase), so are you sure it is `OS400` (uppercase) here? It's a string comparison, so case matters.

**Comment by zheddie** [comment]
@rgommers  , Hi Ralf, I refined the code. Wish this could help to fix the auto-checkk failure. How can I trigger the auto-check again?  thanks.

**Review by rgommers — APPROVED** [review]

**Comment by rgommers** [comment]
Okay, I pushed some tweaks - this LGTM now, in it goes. Thanks @zheddie

**Comment by rgommers** [comment]
New version released: https://pypi.org/project/oldest-supported-numpy/2022.11.19

**Pull request merged**